src/cov.jl

```diff
@@ -48,76 +49,60 @@ function scattermat end
 
 
 """
-    cov(X, wv::WeightVec; mean=nothing, vardim=1)
+    cov(X, wv::AbstractWeights; vardim=1, corrected=false)
 
-Compute the weighted covariance matrix. By default, the covariance
-matrix is normalized by the sum of the weights. That is, `cov(X, wv)`
-is equivalent to `scattermat(X, wv) / sum(wv)`.
+Compute the weighted covariance matrix. Similar to `var` and `std` the biased covariance
+matrix (`corrected=false`) can be computed by multiplying `scattermat(X, wv)` by
+``\\frac{1}{\\sum{w}}`` to normalize. However, the unbiased covariance matrix
+(`corrected=true`) is dependent on the type of weights used:
+
+* AnalyticWeights: ``\\frac{1}{\\sum w - \\sum {w^2} / \\sum w}``
+* FrequencyWeights: ``\\frac{1}{\\sum{w} - 1}``
+* ProbabilityWeights: ``\\frac{n}{(n - 1) \\sum w}`` where ``n`` equals `count(!iszero, w)`
 """
 cov
 
 
 """
-    mean_and_cov(x, [wv::WeightVec]; vardim=1) -> (mean, cov)
+    mean_and_cov(x, [wv::AbstractWeights]; vardim=1, corrected=false) -> (mean, cov)
 
 Return the mean and covariance matrix as a tuple. A weighting
 vector `wv` can be specified. `vardim` that designates whether
 the variables are columns in the matrix (`1`) or rows (`2`).
+Finally, bias correction will be applied to the covariance calculation if
+`corrected=true`. See [`cov`](@ref) documentation for more details.
 """
 function mean_and_cov end
 
 
-@static if VERSION < v"0.5.0-dev+679"
-    function scattermat(x::DenseMatrix; mean=nothing, vardim::Int=1)
-        mean == 0 ? scattermat_zm(x, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, vardim), vardim) :
-        scattermat_zm(x .- mean, vardim)
-    end
-
-    function scattermat(x::DenseMatrix, wv::WeightVec; mean=nothing, vardim::Int=1)
-        mean == 0 ? scattermat_zm(x, wv, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, wv, vardim), wv, vardim) :
-        scattermat_zm(x .- mean, wv, vardim)
-    end
+scattermatm(x::DenseMatrix, mean, vardim::Int=1) =
+    scattermat_zm(x .- mean, vardim)
 
-    ## weighted cov
-    Base.cov(x::DenseMatrix, wv::WeightVec; mean=nothing, vardim::Int=1) =
```

Comment:
Do not remove this method. Also, the docstring needs to mention `mean=nothing` again.